test/lib/getNextJobs.test.js

```diff
@@ -20,15 +20,23 @@ describe('getNextJobs', () => {
         );
     });
 
-    it('should figure out what jobs start next', () => {
-        // trigger for a stage setup with the startFrom as a stage setup
+    it.only('should figure out what jobs start next', () => {
+        // trigger for a stage setup with the startFrom as a stage setup job
         assert.deepEqual(
             getNextJobs(WORKFLOW, {
                 trigger: 'stage@integration:setup',
                 startFrom: 'ci-test'
             }),
             ['ci-test']
         );
+        // trigger for a stage setup with the startFrom as a stage
```

Comment:
Add a test for below scenarios

1. `startFrom` is a stage setup and `trigger` is stage setup, and they belong to the same stage
2. `startFrom` is a stage job and `trigger` is stage setup, but belong to different stages
3. `startFrom` is a stage setup and `trigger` is stage setup, but they belong to different stages